lib/peer.js

```diff
@@ -177,10 +177,11 @@ Peer.prototype.sendMessage = function(message) {
 Peer.prototype.sendExtendedMessage = function(type, data) {
 
   LOGGER.debug('Peer [%s] sending extended message of type %j', this.getIdentifier(), type);
-
   var code = Message.EXTENDED_HANDSHAKE === type 
               ? 0 
-              : this._extensionData && this._extensionData.m[type];
+              : this._extensionData.m !== undefined 
+                  ? this._extensionData && this._extensionData.m[type]
+                  : undefined
```

Comment:
Hey, thanks for the PR! This project isn't really getting any love so nice to have contributions :).

This change could fail if `this._extensionData` is undefined - something more like `this._extensionData && this._extensionData.m && this._extensionData.m[type]` might work. Though if `m` can be undefined it will also be an issue in `Peer.prototype.supportsExtension`. It could be worth add a new function `Peer.prototype.getExtension` which returns they extension value for a key and encapsulates the undefined checking.

Reply:
Oh, ok, Im new in `contributions`. Im learning and liked for this repo. I love .torrents!

Reply:
What do when this._extensionData.m is undefined?

Reply:
Just set code to undefined. It'd be best to break this out so it's more readable. So the code could be - 
```
var code = null;
if (Message.EXTENDED_HANDSHAKE === type) {
  code = 0;
} else if (this._extensionData && this._extensionData.m) {
  code = this._extensionData.m[type];
}
```